crates/turbopack-ecmascript/src/analyzer/builtin.rs

```diff
@@ -501,43 +503,100 @@ pub fn replace_builtin(value: &mut JsValue) -> bool {
         // Reduce logical expressions to their final value(s)
         JsValue::Logical(_, op, ref mut parts) => {
             let len = parts.len();
-            for (i, part) in take(parts).into_iter().enumerate() {
+            let input_parts: Vec<JsValue> = take(parts);
+            *parts = Vec::with_capacity(len);
+            let mut part_properties = Vec::with_capacity(len);
+            for (i, part) in input_parts.into_iter().enumerate() {
                 // The last part is never skipped.
                 if i == len - 1 {
+                    // We intentionally omit the part_properties for the last part.
+                    // This isn't always needed so we only compute it when actually needed.
                     parts.push(part);
                     break;
                 }
-                // We might know at compile-time if a part is skipped or the final value.
-                let skip_part = match op {
+                let property = match op {
                     LogicalOperator::And => part.is_truthy(),
                     LogicalOperator::Or => part.is_falsy(),
                     LogicalOperator::NullishCoalescing => part.is_nullish(),
                 };
-                match skip_part {
+                // We might know at compile-time if a part is skipped or the final value.
+                match property {
                     Some(true) => {
                         // We known this part is skipped, so we can remove it.
                         continue;
                     }
                     Some(false) => {
                         // We known this part is the final value, so we can remove the rest.
+                        part_properties.push(property);
                         parts.push(part);
                         break;
                     }
                     None => {
                         // We don't know if this part is skipped or the final value, so we keep it.
+                        part_properties.push(property);
                         parts.push(part);
                         continue;
                     }
                 }
             }
+            part_properties.truncate(parts.len());
```

Comment:
under what condition does this happen? it seems from the code that part properties is always less than or equal to parts

Reply:
never, removed it...